File: src/main/java/waterpower/client/gui/ContainerFullInventory.java
package waterpower.client.gui;

import net.minecraft.entity.player.EntityPlayer;
import net.minecraft.inventory.Slot;
import waterpower.common.block.tileentity.TileEntityInventory;

public class ContainerFullInventory extends ContainerBase {

    public ContainerFullInventory(EntityPlayer entityPlayer, TileEntityInventory base, int height) {
        super(base);

        for (int row = 0; row < 3; row++) {
            for (int col = 0; col < 9; col++) {
                addSlotToContainer(new Slot(entityPlayer.inventory, col + row * 9 + 9, 8 + col * 18, height + -82 + row * 18));
            }

        }

        for (int col = 0; col < 9; col++)
            addSlotToContainer(new Slot(entityPlayer.inventory, col, 8 + col * 18, height + -24));

    }

}

File: src/main/kotlin/waterpower/common/block/container/ContainerFullInventory.kt
/**
 * Copyright (c) Huang Yuhui, 2017
 *
 * "WaterPower" is distributed under the terms of the Minecraft Mod Public
 * License 1.0, or MMPL. Please check the contents of the license located in
 * http://www.mod-buildcraft.com/MMPL-1.0.txt
 */
package waterpower.common.block.container

import net.minecraft.entity.player.EntityPlayer
import net.minecraft.inventory.IInventory

abstract class ContainerFullInventory : ContainerBase {
    constructor(player: EntityPlayer, inv: IInventory, height: Int) : super(inv) {
        addPlayerInventorySlots(player, height)
    }

    constructor(player: EntityPlayer, inv: IInventory, width: Int, height: Int) : super(inv) {
        addPlayerInventorySlots(player, width, height)
    }
}